src/manifests_workflow/input_manifests_opensearch.py

```
@@ -0,0 +1,99 @@
+            branches = ComponentOpenSearchMin.branches()
+            logging.info(f"Checking OpenSearch {branches} branches")
+            for branch in branches:
+                c = ComponentOpenSearchMin.checkout(
+                    path=os.path.join(work_dir, f"OpenSearch/{branch}"), branch=branch
+                )
+                c.publish_to_maven_local()
+                version = c.version
+                logging.info(f"OpenSearch#{branch} is version {version}")
+                if version not in main_versions.keys():
+                    main_versions[version] = [c]
+
+            # components can increment their own version first without incrementing min
+            manifest = self.latest
+            for component in manifest.components:
+                if component.name == "OpenSearch":
+                    continue
+
+                logging.info(f"Checking out {component.name}#main")
+                component = ComponentOpenSearch.checkout(
+                    name=component.name,
+                    path=os.path.join(work_dir, component.name),
+                    version=manifest.build.version,
+                    branch="main",
+                )
+
+                try:
+                    component_version = component.version
+                    if component_version:
+                        release_version = ".".join(component_version.split(".")[:3])
+                        if release_version not in main_versions.keys():
+                            main_versions[release_version] = []
+                        main_versions[release_version].append(component)
+                        logging.info(
+                            f"{component.name}#main is version {release_version} (from {component_version})"
+                        )
+                except subprocess.CalledProcessError as err:
+                    logging.warn(
+                        f"Error getting version of {component.name}: {str(err)}, ignored"
```

This happen enough that its worth swallowing? Seems like we would want to error out